dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientChannels.java

```diff
@@ -127,12 +140,27 @@ public static Channel create(ClientConfiguration conf) {
                             .register(AuthSchemes.BASIC, new BasicSchemeFactory())
                             .build());
         });
-        CloseableHttpClient client = builder.build();
-        ImmutableList<Channel> channels = conf.uris().stream()
-                .map(uri -> BlockingChannelAdapter.of(new ApacheHttpClientBlockingChannel(client, url(uri))))
-                .collect(ImmutableList.toImmutableList());
 
-        return Channels.create(channels, conf);
+        return new CloseableClient(builder.build());
+    }
+
+    /** Intentionally opaque wrapper type - we don't want people using the inner Apache client directly. */
+    public static final class CloseableClient implements Closeable {
```

Comment:
why is this public if it's never used outside this class?

Reply:
It needs to be public because we create some of these in Witchcraft and then re-use them to avoid re-creating clients on every url live-reload.  

See https://internal-github/foundry/witchcraft/pull/2233/files#diff-2819734323a961448d10003f575f1b4bR217